### ISSUE TITLE
Break up module GenerateROIStatistics 

### ISSUE THREAD

**Issue by dcwhite** [SCIInstitute/SCIRun#623]
Should be separated into generate, filter, and display modules. This will remove the need for the hacky additionalAlgoOutput code.

**Comment by dcwhite** [comment]
@moritzdannhauer, this one might be good for @shayestehfard if you want her to get more module coding experience.

**Comment by moritzdannhauer** [comment]
Yes, after she finished the ones that have been assigned to her before.
I guess she also might need some guidance from you regarding this.

**Comment by dcwhite** [comment]
**Triage 2026-06:** Break up `GenerateROIStatistics` — the module mixes region-of-interest generation, filtering, and display. Splitting into separate modules would improve composability. Medium effort; depends on having users who need the split functionality. The current module works; this is a design improvement.
